torchgeo/datasets/geo.py

```diff
@@ -384,6 +385,23 @@ def dtype(self) -> torch.dtype:
         else:
             return torch.long
 
+    @property
+    def resampling(self) -> Resampling:
+        """Resampling algorithm used when reading input files.
+
+        Defaults to cubic for float dtypes and nearest for int dtypes.
+
+        Returns:
+            The resampling method to use.
+
+        .. versionadded:: 0.6
+        """
+        # Based on torch.is_floating_point
+        if self.dtype in [torch.float64, torch.float32, torch.float16, torch.bfloat16]:
```

Comment:
`torch.double == torch.float64`, and the tests are designed to confirm that both work.